nbtlib/path.py

```diff
@@ -0,0 +1,265 @@
+
+        return '.'.join(filter(None, segments))
+
+
+def parse_accessors(path):
+    try:
+        parser = Parser(tokenize(path))
+    except InvalidLiteral:
+        return ()
+
+    while True:
+        try:
+            tag = parser.parse()
+        except InvalidLiteral as exc:
+            raise InvalidPath(f'Invalid path at position {exc.args[0][0]}') from exc
+
+        if isinstance(tag, String):
+            if parser.current_token.type == 'QUOTED_STRING':
+                yield NamedKey(tag[:])
+            else:
+                yield from (NamedKey(key) for key in tag.split('.') if key)
+
+        elif isinstance(tag, List):
+            if not tag:
+                yield ListIndex(index=None)
+            elif len(tag) != 1:
+                raise InvalidPath('Brackets should only contain one element')
+            elif issubclass(tag.subtype, Int):
+                yield ListIndex(int(tag[0]))
+            elif issubclass(tag.subtype, Compound):
+                yield ListIndex(index=None)
+                yield CompoundMatch(tag[0])
+            else:
+                raise InvalidPath('Brackets should only contain an integer or a compound')
+
+        elif isinstance(tag, Compound):
+            yield CompoundMatch(tag)
+
+        else:
+            raise InvalidPath(f'Invalid path element {tag}')
+
+        try:
+            parser.next()
+        except InvalidLiteral:
+            break
+
+
+def extend_accessors(accessors, new_accessor):
+    if isinstance(new_accessor, CompoundMatch) and accessors:
+        *except_last, last_accessor = accessors
+
+        if isinstance(last_accessor, CompoundMatch):
+            return tuple(except_last) + (
+                CompoundMatch(new_accessor.compound.with_defaults(last_accessor.compound)),
+            )
+        if isinstance(last_accessor, ListIndex) and last_accessor.index is not None:
+            raise InvalidPath('Can\'t match a compound on list items '
+                              f'selected with {last_accessor!r}')
+    return accessors + (new_accessor,)
+
+
+class NamedKey(NamedTuple):
+    key: str
+
+    UNQUOTED_REGEX = re.compile(r'^[a-zA-Z0-9_]+$')
+
+    def get(self, tags):
+        return [(tag, tag[self.key]) for _, tag in tags if self.key in tag]
+
+    def set(self, tags, value):
+        for parent, _ in tags:
+            parent[self.key] = value
+
+    def delete(self, tags):
+        for parent, _ in tags:
+            if self.key in parent:
+                del parent[self.key]
+
+    def __str__(self):
+        return (
+            self.key if self.UNQUOTED_REGEX.match(self.key) else
+            '"' + self.key.replace('"', '\\"') + '"'
+        )
+
+
+class ListIndex(NamedTuple):
+    index: Optional[int]
+
+    def get(self, tags):
+        if self.index is None:
+            return [((tag, i), item)
+                    for _, tag in tags
+                    for i, item in enumerate(tag)]
+        return [((tag, self.index), tag[self.index])
```

Comment:
Elements got from Int/Byte/LongArray is pure `int`.
it should be Int/Byte/Long,respectively.

Reply:
Hmm this is because arrays are [numpy arrays](https://github.com/vberlier/nbtlib/blob/master/nbtlib/tag.py#L313) and I'm not overriding `__getitem__`, even on master. Maybe this could be a separate issue?

Reply:
what is faster?
`int.__new__(Int,5)` or `Int(5)`?
as we can make sure that the number get from the Array is in the range.

Reply:
after some testing,int.__new__(Int,5) is much faster than Int(5).
we should use that, as we dont need to test if the number get from the Array is in the range.

Reply:
Sounds good, I'll fix this directly on master, thanks :+1:

Reply:
Done 2ba23ae